drv/stm32h7-fmc-demo-server/src/main.rs

```diff
@@ -226,15 +251,57 @@ fn main() -> ! {
 
 struct ServerImpl {
     fmc: &'static device::fmc::RegisterBlock,
+    net: Net,
+    rx_packet: &'static mut [u8; 1024],
+    tx_packet: &'static mut [u8; 1024],
 }
 
 impl NotificationHandler for ServerImpl {
     fn current_notification_mask(&self) -> u32 {
-        0
+        notifications::SOCKET_MASK
     }
 
-    fn handle_notification(&mut self, _bits: u32) {
-        unreachable!()
+    fn handle_notification(&mut self, bits: u32) {
+        if bits & notifications::SOCKET_MASK != 0 {
```

Comment:
**suggestion:** Invert the condition and do an early return to get one less level of indentation for the loop?

Reply:
A good suggestion in general -- I like early returns -- but in case this notification handler should grow a second notification in the future, I'd rather follow this pattern, so that the second bit we need to test causes less churn.